### PR TITLE
Add information on building RDkit on macOS using conda

### PR DESCRIPTION
**Pull request by gedeck**
Specific information for setting up the environment, building and testing. 

Ideally, we should fix the CMakeCache.txt file somewhere else.

### CODE REVIEW

**Comment by greglandrum** [comment]
@gedeck I agree that we should update that documentation, but I'm not sure that this the best approach for doing that. Certainly one doesn't need to edit CMakeCache.txt.

I don't have time right now to propose a real alternative, but the attached script is the one that I use to configure a build directory on my Mac using anaconda python.
[build_conda_release.sh.txt](https://github.com/rdkit/rdkit/files/608341/build_conda_release.sh.txt)

File: Docs/Book/Install.md [hunk 1]
```diff
@@ -33,7 +33,40 @@ Windows users will use a slightly different command:
...
+For more details on building from source with Conda, see the [conda-rdkit repository](https://github.com/rdkit/conda-rdkit).
+
+#### macOS 10.12 (Sierra): Python 3 environment
+
+The following commands will create a development environment for macOS Sierra and Python 3. Download
+Miniconda3-latest-MacOSX-x86_64.sh from [Conda](http://conda.pydata.org/miniconda.html) and run these
+following commands:
+   
+	bash Miniconda3-latest-MacOSX-x86_64.sh
+	conda install numpy scipy matplotlib
```
Comment (greglandrum) [inline Docs/Book/Install.md]: I don't think scipy is required

File: Docs/Book/Install.md [hunk 2]
```diff
@@ -33,7 +33,40 @@ Windows users will use a slightly different command:
...
+	conda install cmake
+	conda install -c  https://conda.anaconda.org/rdkit  boost
```
Comment (greglandrum) [inline Docs/Book/Install.md]: `conda -c rdkit boost` is good enough

File: Docs/Book/Install.md [hunk 3]
```diff
@@ -33,7 +33,40 @@ Windows users will use a slightly different command:
...
+	conda install -c  https://conda.anaconda.org/rdkit  cairocffi
```
Comment (greglandrum) [inline Docs/Book/Install.md]: `conda install -c cairocffi` is good enough

File: Docs/Book/Install.md [hunk 4]
```diff
@@ -33,7 +33,40 @@ Windows users will use a slightly different command:
...
+	conda install pillow
+	conda install anaconda
+	conda install --channel https://conda.anaconda.org/rwest eigen3
```
Comment (greglandrum) [inline Docs/Book/Install.md]: I have been installing eigen from conda-forge.
Given that this is somewhat more "official" than rwest's repo, I would suggest using that:
`conda install -c conda-forge eigen`

File: Docs/Book/Install.md [hunk 5]
```diff
@@ -33,7 +33,40 @@ Windows users will use a slightly different command:
...
+Optionally, add the following packages to your environment. 
+
+	pip install yapf==0.11.1
+	pip install coverage==3.7.1
+
+Then follow the usual build instructions. The PYTHON\_INCLUDE\_DIR must be set in the
+cmake command.
+
+	PYROOT=<path to miniconda3>
+	cmake -DPYTHON_INCLUDE_DIR=$PYROOT/include/python3.5m ..
```
Comment (greglandrum) [inline Docs/Book/Install.md]: This is way more minimal than my cmake command. Are you sure that this works?

File: Docs/Book/Install.md (hunk 2 above)
Comment (gedeck) [inline Docs/Book/Install.md]: I remove the full URL here and for cairocffi and eigen.

File: Docs/Book/Install.md (hunk 3 above)
Comment (gedeck) [inline Docs/Book/Install.md]: Done

File: Docs/Book/Install.md (hunk 4 above)
Comment (gedeck) [inline Docs/Book/Install.md]: Changed (and changed my environment too)

File: Docs/Book/Install.md (hunk 5 above)
Comment (gedeck) [inline Docs/Book/Install.md]: This is yours:
```
cmake -DPYTHON_EXECUTABLE=$PYROOT/bin/python  \
          -DRDK_BUILD_AVALON_SUPPORT=ON \
          -DRDK_BUILD_INCHI_SUPPORT=ON \
          -DRDK_BUILD_CAIRO_SUPPORT=ON -DRDK_BUILD_THREADSAFE_SSS=ON -DRDK_TEST_MULTITHREADED=ON \
	  -DPYTHON_LIBRARY=$PYROOT/lib/libpython3.5m.dylib \
	  -DPYTHON_INCLUDE_DIR=$PYROOT/include/python3.5m \
	  -DPYTHON_NUMPY_INCLUDE_PATH=$PYROOT/lib/python3.5/site-packages/numpy/core/include/ \
	  -DCMAKE_BUILD_TYPE=Release \
	  -DRDK_INSTALL_INTREE=OFF \
	  -DCMAKE_INSTALL_PREFIX=$INSTALL \
          -DBOOST_ROOT=$PYROOT \
          -DBoost_NO_BOOST_CMAKE=FALSE \
          -DBoost_NO_SYSTEM_PATHS=on \
          -DBoost_USE_STATIC_LIBS=off \
	  $SRCDIR 
```
I just checked what the settings are in the CMakeCache.txt file after running my command:

- `PYTHON_EXECUTABLE`: correctly found
- `PYTHON_LIBRARY`: correctly found
- `PYTHON_INCLUDE_DIR`: that is not correctly found and needs to be specified
- `PYTHON_NUMPY_INCLUDE_PATH`: correctly found
- `RDK_BUILD_AVALON_SUPPORT`: now added
- `RDK_BUILD_INCHI_SUPPORT`: now added
- `RDK_BUILD_CAIRO_SUPPORT`: now added 
- `RDK_BUILD_THREADSAFE_SSS`: on by default
- `RDK_TEST_MULTITHREADED`: on by default
- `CMAKE_BUILD_TYPE`: default
- `RDK_INSTALL_INTREE`: ON by default and that works for me
- `CMAKE_INSTALL_PREFIX`: not required for RDK_INSTALL_INTREE

Then there are the Boost settings. I cannot find these in the CMakeCache.txt file.

- `BOOST_ROOT`
- `Boost_NO_BOOST_CMAKE`
- `Boost_NO_SYSTEM_PATHS`
- `Boost_USE_STATIC_LIBS

File: Docs/Book/Install.md [hunk 6]
```diff
@@ -33,7 +33,43 @@ Windows users will use a slightly different command:
...
+	pip install yapf==0.11.1
+	pip install coverage==3.7.1
+
+Then follow the usual build instructions. The PYTHON\_INCLUDE\_DIR must be set in the
+cmake command.
+
+	PYROOT=<path to miniconda3>
+	cmake -DPYTHON_INCLUDE_DIR=$PYROOT/include/python3.5m  \
+      -DRDK_BUILD_AVALON_SUPPORT=ON \
+      -DRDK_BUILD_CAIRO_SUPPORT=ON \
```
Comment (greglandrum) [inline Docs/Book/Install.md]: I think the cairo support will only work if you've also installed cairo manually (something of a pain in the butt) or also install it via conda.

**Comment by greglandrum** [comment]
This one is just waiting for the cairo install lines to be added and then it's ready to go

**Comment by greglandrum** [comment]
Thanks Peter!

**Pull request merged**